string name
int32 id
int32 flag_query
